### PR TITLE
ci(lint): added `autocorrect`

### PR DESCRIPTION
**Pull request by 3fuyang**
## What type of PR is this?

- [x] Enhancement
- [ ] Displaying
- [x] Typo
- [ ] Doc Request

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

- feat(lint): added [autocorrect](https://github.com/huacnlee/autocorrect) in order to enhance the linting process for Chinese language content in the codebase
- perf(lint): lint only the changed files in CI
- build(node): updated `node` to `20.9.0 LTS`
- build(package-manager): migrated to `pnpm`
- ci(lint): triggers only on PRs

### CODE REVIEW

**Review by dengn — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,36 +1,55 @@
...
     runs-on: ubuntu-latest
     steps:
     - name: Check out code
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       with:
         repository: ${{ github.event.pull_request.head.repo.full_name }}
         ref: ${{ github.event.pull_request.head.ref }}
+        # Assume PRs are less than 20 commits
+        fetch-depth: 20
```
Comment (xuoutput) [inline .github/workflows/lint.yml:17]: 用默认的 1 层 depth 就够了

File: package-lock.json
Comment (xuoutput) [inline package-lock.json:1]: 怎么有 lock.json ,又有 lock. yaml

Comment (3fuyang) [reply package-lock.json:1]: 旧的 CI 里 install deps 部分用的 `npm ci`，先保留 `lock.json` 保证能通过 check，再把 `package-lock.json` 移除，后面统一用 `pnpm`

File: .autocorrectrc [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  # Config the files associations, you config is higher priority than default.
+  # "rb": ruby
+  # "Rakefile": ruby
+  # "*.js": javascript
+  # ".mdx": markdown
+spellcheck:
+  words:
+    # Please do not add a general English word (eg. apple, python) here.
+    # Users can add their special words to their .autocorrectrc file by their need.
+    - ActiveMQ
```
Comment (xuoutput) [inline .autocorrectrc:21]: 到时候是要在这里补充 MO 专属名词吧

Comment (3fuyang) [reply .autocorrectrc:21]: 对

File: package.json [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+  "packageManager": "pnpm@8.10.2",
   "scripts": {
     "config:zhlint": "node ./scripts/genZhlintConfig",
-    "lint": "npm run config:zhlint && npx zhlint './docs/MatrixOne/**/*.md' && npx markdownlint-cli2 './docs/MatrixOne/**/*.md'",
-    "lint:fix": "npm run config:zhlint && npx zhlint './docs/MatrixOne/**/*.md' --fix && npx markdownlint-cli2-fix './docs/MatrixOne/**/*.md'"
+    "lint": "npm run lint:autocorrect && npm run lint:zhlint && npm run lint:markdownlint",
+    "lint:fix": "npm run lint:autocorrect:fix && npm run lint:zhlint:fix && npm run lint:markdownlint:fix",
+    "lint:autocorrect": "node ./scripts/autocorrect",
+    "lint:autocorrect:fix": "node ./scripts/autocorrect --fix",
+    "lint:zhlint": "npm run config:zhlint && npx zhlint './docs/MatrixOne/**/*.md'",
```
Comment (xuoutput) [inline package.json:11]: 后面会用 autocorrect 代替 zhlint 的么

**Review by xuoutput — APPROVED** [review]
LGTM

File: package.json (hunk 3 above)
Comment (3fuyang) [reply package.json:11]: 打算先观察一段时间效果如何，再决定要不要全面替换

**Pull request merged**